### ISSUE TITLE
How to check on how many genes the tree is based

### ISSUE THREAD

**Issue by MichaelsGITIGIT** [molbiodiv/bcgTree#20]
What is the most convenient way to check on how many genes the tree is based?

Do I have to check in the absence_presence.csv for which TIGR number it found a hit for every bacterium?

Or is it somewhere in the log?

**Comment by iimog** [comment]
In general all genes are used that are found in more than one genome. For the genomes that lack this gene lines with gap only are inserted in the final alignment (after Gblocks). So in general the tree is based on all genes with at least two hits. The log contains a line for each gene with at least one hit like this:
```
[11-06 10:30:49] [bcgTree] Wrote 9 ids for gene TIGR01021
```
And if the gene has only one hit an additional line like this:
```
[11-06 10:30:49] [bcgTree] Gene TIGR02387 only present in 1 proteome - removing from further analyses
```
So if you are on Linux or OS X you can use the following commands to quickly determine the respective numbers:
```bash
# Number of genes with at least one hit
grep "Wrote" bcgTree.log | wc -l
# Number of genes with exactly one hit (excluded)
grep "removing" bcgTree.log | wc -l
```

**Comment by MichaelsGITIGIT** [comment]
But isn't RAxML ignoring all columns with gaps anyway?

**Comment by iimog** [comment]
As I understand the following section from [the manual](http://sco.h-its.org/exelixis/resource/download/NewManual.pdf) gap columns are not entirely ignored:
> Q: How does RAxML handle gaps
> A: Like   most   other   likelihood-based   phylogeny   programs   it   handles   them   as   undetermined characters, that is gaps are treated as Ns. If this is new to you you may want to read Joe Felsenstein's textbook Inferring Phylogenies where he explains why this is a reasonable way to model gaps.

Anyway would it be useful for you to have an option to only use genes that are found in all of the genomes?

**Comment by MichaelsGITIGIT** [comment]
Oh, OK, thanks for this clarification!

Yes, this might be a nice feature! Guess not only for me...

**Comment by MichaelsGITIGIT** [comment]
Hmmm.... I see from #21 that it would be possible using the command line. Just used the GUI so far, but I guess then I will switch to command line use anyway. But I still think it would be nice to add this feature, since the implementation seems to be reasonable easy by exploiting the threshold parameter...
Thx a lot!

**Comment by iimog** [comment]
I just released a new version (1.0.9) that includes the parameters in the command line interface. I also included those options in the GUI under "Advanced settings" so no need to switch to the CLI. Hope that helps. Thanks for your help in improving bcgTree.

**Comment by MichaelsGITIGIT** [comment]
Great! Many thanks.
